### PR TITLE
Add --all-lists to query command

### PR DESCRIPTION
**Pull request by jorgectf**
Changed `--all` to `--all-projects` to query all followed projects, and added `--all-lists` to query all created lists.

### CODE REVIEW

File: cli.go [hunk 1]
```diff
@@ -824,6 +828,13 @@ func main() {
 						Fatalf("file is not a .ql: %s", queryFilepath)
 					}
 
+					projectListKeys := c.StringSlice("list-key")
+					projectListNames := c.StringSlice("list")
+					allListsFlag := c.Bool("all-lists")
+					if len(projectListKeys)+len(projectListNames) > 0 && allListsFlag {
```
Comment (jorgectf) [inline cli.go]: Haven't checked `--all-projects` since project lists may contain non-followed but built projects.

File: cli.go [hunk 2]
```diff
@@ -803,9 +803,13 @@ func main() {
 						Usage: "Filepath to text file with list of repos.",
 					},
 					&cli.BoolFlag{
-						Name:  "all, a",
+						Name:  "all-projects, ap",
 						Usage: "Query all followed projects.",
 					},
+					&cli.BoolFlag{
+						Name:  "all-lists, al",
+						Usage: "Query all created lists.",
```
Comment (gagliardetto) [inline cli.go]:
```suggestion
						Usage: "Query all current user's lists.",
```

File: cli.go [hunk 3]
```diff
@@ -967,21 +978,27 @@ func main() {
...
 						if err != nil {
 							panic(err)
 						}
 
-						for _, name := range projectListNames {
-							list := lists.ByName(name)
-							if list == nil {
-								Warnf("List %q not found; skipping", name)
-							} else {
+						if len(projectListNames) > 0 {
```
Comment (gagliardetto) [inline cli.go]: This check is kinda redundant; the loop won't iterate if the slice is empty.

```
						if len(projectListNames) > 0 {
```

File: cli.go [hunk 4]
```diff
@@ -803,9 +803,13 @@ func main() {
 						Usage: "Filepath to text file with list of repos.",
 					},
 					&cli.BoolFlag{
-						Name:  "all, a",
+						Name:  "all-projects, ap",
```
Comment (gagliardetto) [inline cli.go]:
```suggestion
						Name:  "all-followed, af",
```

File: cli.go [hunk 5]
```diff
@@ -888,7 +899,7 @@ func main() {
 							// With cache:
 
 							// If no repos specified, and flag --all is true, then query all:
-							if c.Bool("all") {
+							if c.Bool("all-projects") {
```
Comment (gagliardetto) [inline cli.go]:
```suggestion
							if c.Bool("all-followed") {
```

File: cli.go [hunk 6]
```diff
@@ -824,6 +828,13 @@ func main() {
 						Fatalf("file is not a .ql: %s", queryFilepath)
 					}
 
+					projectListKeys := c.StringSlice("list-key")
+					projectListNames := c.StringSlice("list")
+					allListsFlag := c.Bool("all-lists")
```
Comment (gagliardetto) [inline cli.go]:
```suggestion
					doAllLists := c.Bool("all-lists")
```

File: cli.go (hunk 1 above)
Comment (gagliardetto) [inline cli.go]:
```suggestion
					if len(projectListKeys)+len(projectListNames) > 0 && doAllLists {
```

File: cli.go [hunk 7]
```diff
@@ -967,21 +978,27 @@ func main() {
 						}
 					}
 
-					projectListKeys := c.StringSlice("list-key")
-					projectListNames := c.StringSlice("list")
-
-					if len(projectListNames) > 0 {
-						// Add project lists by name:
+					if len(projectListNames) > 0 || allListsFlag {
```
Comment (gagliardetto) [inline cli.go]:
```suggestion
					if len(projectListNames) > 0 || doAllLists {
```

File: cli.go [hunk 8]
```diff
@@ -967,21 +978,27 @@ func main() {
...
+								list := lists.ByName(name)
+								if list == nil {
+									Warnf("List %q not found; skipping", name)
+								} else {
+									projectListKeys = append(projectListKeys, list.Key)
+								}
+							}
+						}
+
+						if allListsFlag {
```
Comment (gagliardetto) [inline cli.go]:
```suggestion
						if doAllLists {
```

**Review by gagliardetto — CHANGES_REQUESTED** [review]

File: cli.go (hunk 3 above)
Comment (jorgectf) [reply cli.go]: Right! Didn't realize it :O

Comment (jorgectf) [reply cli.go]: Done https://github.com/gagliardetto/lgtm-cli/pull/5/commits/7259ceca8bc0a9aec8247efbcf09de3fa880d5aa :P

**Comment by jorgectf** [comment]
Apparently force-pushed before pulling the commit done by the reviews (I thought I did lol) and then the commit didn't exist for `git reset`...

File: cli.go [hunk 9]
```diff
@@ -967,21 +978,23 @@ func main() {
...
-
-					if len(projectListNames) > 0 {
-						// Add project lists by name:
+					if len(projectListNames) > 0 || doAllLists {
 						lists, err := client.ListProjectSelections()
 						if err != nil {
 							panic(err)
 						}
 
-						for _, name := range projectListNames {
```
Comment (gagliardetto) [inline cli.go:980]: This range loop is needed.

**Review by gagliardetto — CHANGES_REQUESTED** [review]

File: cli.go (hunk 9 above)
Comment (jorgectf) [reply cli.go:980]: Sometimes I doubt if there's an evil mastermind controlling me while committing... 😂 will get that loop back tomorrow 👍

**Comment by jorgectf** [comment]
Evil mastermind defeated 😆

**Comment by gagliardetto** [comment]
Great! Thanks :relaxed:

**Pull request merged**